client/dashboard/src/components/sources/ViewAssetDialogContent.tsx

```diff
@@ -0,0 +1,70 @@
+import { CodeBlock } from "@/components/code";
+import { SkeletonCode } from "@/components/ui/skeleton";
+import { useProject } from "@/contexts/Auth";
+import { getServerURL } from "@/lib/utils";
+import { Dialog } from "@speakeasy-api/moonshine";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router";
+import { NamedAsset } from "./SourceCard";
+
+interface ViewAssetDialogContentProps {
+  asset: NamedAsset;
+}
+
+export function ViewAssetDialogContent({ asset }: ViewAssetDialogContentProps) {
+  const { projectSlug } = useParams();
+  const project = useProject();
+  const [content, setContent] = useState<string>("");
+  const [isPending, setIsPending] = useState(true);
+
+  const downloadURL = new URL("/rpc/assets.serveOpenAPIv3", getServerURL());
+  downloadURL.searchParams.set("id", asset.id);
+  downloadURL.searchParams.set("project_id", project.id);
+
+  useEffect(() => {
+    if (!projectSlug) {
+      setContent("");
+      return;
+    }
+
+    setIsPending(true);
+    fetch(downloadURL, {
+      credentials: "same-origin",
+    })
+      .then((assetData) => {
+        if (!assetData.ok) {
+          setContent("");
+          setIsPending(false);
+          return;
+        }
+        return assetData.text();
+      })
+      .then((content) => {
+        if (content) {
+          setContent(content);
+        }
+        setIsPending(false);
+      })
+      .catch(() => {
+        setContent("");
+        setIsPending(false);
+      });
```

Comment:
why not use an async function within the effect and use `await` to avoid the `.then()` chaining